<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>meffilTools 2 • meffilTools</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="meffilTools 2">
<meta property="og:description" content="meffilTools">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">meffilTools</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.14</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/meffilTools.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/vjcitn/meffilTools">
    <span class="fas fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="meffilTools_files/header-attrs-2.8/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>meffilTools 2</h1>
                        <h4 class="author">Channing software teams</h4>
            
            <h4 class="date">June 03, 2021</h4>
      
      
      <div class="hidden name"><code>meffilTools.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The meffil package (<span class="citation">Min et al. (2018)</span>) for Illumina Infinium methylation array analysis produces functionally normalized signal estimates in a memory-sparing disk format, GDS.</p>
<p>The GDS format requires additional infrastructure for simple downstream manipulation. We’ve converted all the betas to chromosome-specific HDF5 SummarizedExperiments.</p>
<p>An example HDF5 SummarizedExperiment is provided with the pidsleyEPICData package. The raw data are from NCBI GEO <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE86831">GSE86831</a>, as described in <span class="citation">Pidsley et al. (2016)</span>. <!--, a [2016 Genome Biology
paper](https://doi.org/10.1186/s13059-016-1066-1)
by Pidsley and colleagues.
--></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/HDF5Array">HDF5Array</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">meffilTools</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/PeteHaitch/DelayedMatrixStats">DelayedMatrixStats</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/LTLA/BiocSingular">BiocSingular</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://www.stat.berkeley.edu/~breiman/RandomForests/">randomForest</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">pidsleyEPICData</span><span class="op">)</span>
<span class="op">}</span><span class="op">)</span>

<span class="va">pidsSE</span> <span class="op">=</span> <span class="fu">pidsleyEPICData</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pidsleyEPICData/man/pidsleyData.html">pidsleyData</a></span><span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span>
<span class="va">pidsSE</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 846232 15 
## metadata(0):
## assays(1): betas
## rownames(846232): cg14817997 cg26928153 ... cg07660283 cg09226288
## rowData names(0):
## colnames: NULL
## colData names(9): Sample_Name Sample_Well ... Basename Sex</code></pre>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">pidsSE</span><span class="op">)</span></code></pre></div>
<pre><code>## &lt;846232 x 15&gt; matrix of class DelayedMatrix and type "double":
##                  [,1]       [,2]       [,3] ...     [,14]     [,15]
## cg14817997  0.7789272  0.8155561  0.7164569   . 0.5773467 0.7401255
## cg26928153  0.9467096  0.9481201  0.9028903   . 0.9038258 0.8960001
## cg16269199  0.8725490  0.8803706  0.7796930   . 0.7817547 0.7756640
## cg13869341  0.8909318  0.8654525  0.8421502   . 0.8219929 0.8620466
## cg14008030  0.6627212  0.7052351  0.6830013   . 0.5262791 0.5069644
##        ...          .          .          .   .         .         .
## cg16817496 0.13411024 0.14831004 0.08803999   . 0.1791183 0.2126328
## cg15964525 0.89694635 0.89705107 0.81047270   . 0.8144210 0.8364056
## cg05111475 0.87763399 0.89074500 0.76547307   . 0.7678049 0.8236551
## cg07660283 0.84969323 0.80442862 0.61971544   . 0.7302855 0.7594152
## cg09226288 0.70827453 0.75876353 0.62220802   . 0.4861417 0.6151339</code></pre>
<p>Fifteen arrays are provided in raw form at GSE86831. Biological source types are labeled in <code>Sample_Group</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="va">pidsSE</span><span class="op">$</span><span class="va">Sample_Group</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##     CAF Guthrie   LNCaP     NAF    PrEC 
##       3       5       2       3       2</code></pre>
<p>“CAF” and “NAF” refer to paired samples of tumor-associated or normal fibroblasts. “LNCaP” refers to a transformed prostate cancer cell line. “PrEC” denotes primary cell culture of prostate epithelial cells. “Guthrie” samples are infant blood derived from “archival Guthrie cards.”</p>
</div>
<div id="tasks-supported-by-meffiltools" class="section level1">
<h1 class="hasAnchor">
<a href="#tasks-supported-by-meffiltools" class="anchor"></a>Tasks supported by meffilTools</h1>
<p>Throughout the remainder, we will assume that the default processing of the arrays by meffil produces adequate estimates of “beta” for CpGs on the array.</p>
<div id="compute-the-iccs-for-a-set-of-cpgs" class="section level2">
<h2 class="hasAnchor">
<a href="#compute-the-iccs-for-a-set-of-cpgs" class="anchor"></a>Compute the ICCs for a set of CpGs</h2>
<p>We use <code><a href="https://rdrr.io/pkg/nlme/man/gls.html">nlme::gls</a></code>. A function must be defined to derive a suitable data.frame from the assay results for a feature. An example is shown below.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/HDF5Array">HDF5Array</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></span><span class="op">)</span>
<span class="va">makedf_simple</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">se</span>, <span class="va">...</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">==</span><span class="fl">1</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>beta<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span>,
       id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">se</span><span class="op">$</span><span class="va">Sample_Group</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="../reference/get_iccs.html">get_iccs</a></span><span class="op">(</span><span class="va">pidsSE</span>, inds<span class="op">=</span><span class="fl">51</span><span class="op">:</span><span class="fl">60</span>, makedf.in<span class="op">=</span><span class="va">makedf_simple</span>, fixedfmla <span class="op">=</span> <span class="va">beta</span><span class="op">~</span><span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code>##  cg10037654  cg14057946  cg11422233  cg16047670  cg18263059  cg01014490 
##  0.94134271  0.85556159  0.46110075  0.06809834  0.06478950 -0.25000000 
##  cg11731845  cg04231937  cg04496485  cg24063007 
##  0.09079412  0.05644596  0.67231419 -0.25000000</code></pre>
<p>We can elaborate the GDS model by changing the makedf and fixedfmla.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">makedf_celltype</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">se</span>, <span class="va">...</span><span class="op">)</span> <span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/stopifnot.html">stopifnot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">==</span><span class="fl">1</span><span class="op">)</span>
  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>beta<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">se</span><span class="op">)</span><span class="op">)</span>,
       ngc<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="op">!</span><span class="op">(</span><span class="va">se</span><span class="op">$</span><span class="va">Sample_Group</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Guthrie"</span>, <span class="st">"CAF"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
       id<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">se</span><span class="op">$</span><span class="va">Sample_Group</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="op">}</span>
<span class="fu"><a href="../reference/get_iccs.html">get_iccs</a></span><span class="op">(</span><span class="va">pidsSE</span>, inds<span class="op">=</span><span class="fl">51</span><span class="op">:</span><span class="fl">60</span>, makedf.in<span class="op">=</span><span class="va">makedf_celltype</span>, fixedfmla <span class="op">=</span> <span class="va">beta</span><span class="op">~</span><span class="va">ngc</span><span class="op">)</span></code></pre></div>
<pre><code>##  cg10037654  cg14057946  cg11422233  cg16047670  cg18263059  cg01014490 
##  0.94182586  0.87547233  0.54519607 -0.03365774 -0.25000000 -0.25000000 
##  cg11731845  cg04231937  cg04496485  cg24063007 
##  0.20760962 -0.01463034  0.72962983 -0.25000000</code></pre>
<p>This illustrates the sensitivity of ICC estimation to the model for the mean response.</p>
</div>
<div id="exploration-of-differential-methylation" class="section level2">
<h2 class="hasAnchor">
<a href="#exploration-of-differential-methylation" class="anchor"></a>Exploration of differential methylation</h2>
<p>Selection of useful features can proceed by many avenues.</p>
<p>We’ll use approximate principal components analysis (PCA to 6 components via the “implicitly restarted Lanczos bidiagonalization algorithm” in the irlba package, interfaced in BiocSingular) to look at adequacy of small numbers of probes for discriminating the cell types in the Pidsley dataset.</p>
<div id="filtering-to-the-most-variable-cpg-probes-reporting" class="section level3">
<h3 class="hasAnchor">
<a href="#filtering-to-the-most-variable-cpg-probes-reporting" class="anchor"></a>Filtering to the most variable CpG probes; reporting</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">allm</span> <span class="op">=</span> <span class="fu">rowMads</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">pidsSE</span><span class="op">)</span><span class="op">)</span>
<span class="va">omads</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">allm</span>, decreasing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">top50</span> <span class="op">=</span> <span class="va">pidsSE</span><span class="op">[</span><span class="va">omads</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fl">50</span><span class="op">)</span><span class="op">]</span>,<span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">top50</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 50 15</code></pre>
<p>The <code>rept_filtse</code> function will compute approximate PCA and will use random forests to measure discriminative capacity in selected features.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span>
<span class="va">o50</span> <span class="op">=</span> <span class="fu"><a href="../reference/rept_filtse.html">rept_filtse</a></span><span class="op">(</span><span class="va">top50</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">o50</span><span class="op">)</span></code></pre></div>
<p><img src="meffilTools_files/figure-html/doittttp-1.png" width="700"></p>
<pre><code>## 
## Call:
##  randomForest(formula = y ~ ., data = newdf, importance = TRUE) 
##                Type of random forest: classification
##                      Number of trees: 500
## No. of variables tried at each split: 7
## 
##         OOB estimate of  error rate: 13.33%
## Confusion matrix:
##         CAF Guthrie LNCaP NAF PrEC class.error
## CAF       2       0     0   1    0   0.3333333
## Guthrie   0       5     0   0    0   0.0000000
## LNCaP     0       0     2   0    0   0.0000000
## NAF       1       0     0   2    0   0.3333333
## PrEC      0       0     0   0    2   0.0000000</code></pre>
<p><img src="meffilTools_files/figure-html/doittttp-2.png" width="700"></p>
<p>The biplots show that PrEC, LNCaP, and Guthries cells can be distinguished, and replicates grouped well together, using PC1 and PC2 on 50 probes, but NAF and CAF cells are grouped together.</p>
</div>
<div id="restricting-attention-to-promoter-regions" class="section level3">
<h3 class="hasAnchor">
<a href="#restricting-attention-to-promoter-regions" class="anchor"></a>Restricting attention to “promoter regions”</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pidsSE_p</span> <span class="op">=</span> <span class="fu"><a href="../reference/subset_to_promoters.html">subset_to_promoters</a></span><span class="op">(</span><span class="va">pidsSE</span><span class="op">)</span>
<span class="va">pmads</span> <span class="op">=</span> <span class="fu">rowMads</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">pidsSE_p</span><span class="op">)</span><span class="op">)</span>
<span class="va">opmads</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span><span class="op">(</span><span class="va">pmads</span>, decreasing<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">top50_p</span> <span class="op">=</span> <span class="va">pidsSE_p</span><span class="op">[</span><span class="va">opmads</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq_len</a></span><span class="op">(</span><span class="fl">50</span><span class="op">)</span><span class="op">]</span>,<span class="op">]</span>
<span class="va">o50_p</span> <span class="op">=</span> <span class="fu"><a href="../reference/rept_filtse.html">rept_filtse</a></span><span class="op">(</span><span class="va">top50_p</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">o50_p</span><span class="op">)</span></code></pre></div>
<p><img src="meffilTools_files/figure-html/dopfilt-1.png" width="700"></p>
<pre><code>## 
## Call:
##  randomForest(formula = y ~ ., data = newdf, importance = TRUE) 
##                Type of random forest: classification
##                      Number of trees: 500
## No. of variables tried at each split: 7
## 
##         OOB estimate of  error rate: 6.67%
## Confusion matrix:
##         CAF Guthrie LNCaP NAF PrEC class.error
## CAF       3       0     0   0    0   0.0000000
## Guthrie   0       5     0   0    0   0.0000000
## LNCaP     0       0     2   0    0   0.0000000
## NAF       1       0     0   2    0   0.3333333
## PrEC      0       0     0   0    2   0.0000000</code></pre>
<p><img src="meffilTools_files/figure-html/dopfilt-2.png" width="700"> Some annotation of CpG probes of interest is available in <code>epic_granges</code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">epic_granges</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"cg00451642"</span>, <span class="st">"cg14674796"</span><span class="op">)</span><span class="op">]</span></code></pre></div>
<pre><code>## GRanges object with 2 ranges and 2 metadata columns:
##              seqnames    ranges strand |   isl_class     in_prom
##                 &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt;
##   cg00451642    chr15  66546935      - |     S_Shore      MEGF11
##   cg14674796    chr17  70115831      + |     N_Shore        SOX9
##   -------
##   seqinfo: 24 sequences from hg19 genome; no seqlengths</code></pre>
</div>
<div id="using-a-gene-set" class="section level3">
<h3 class="hasAnchor">
<a href="#using-a-gene-set" class="anchor"></a>Using a gene set</h3>
<p>In this section we use a set of genes identified in <span class="citation">Romanuik et al. (2010)</span> defining an “LNCaP Atlas.” Our anticipation is that CpG probes in these genes will help isolate the LNCaP samples from the others.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">lncap_activ</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"DHRS7"</span>, <span class="st">"GAS5"</span>, <span class="st">"PSMA7"</span>, <span class="st">"ELOVL5"</span>, <span class="st">"ENO2"</span>, <span class="st">"DHCR24"</span>, <span class="st">"FLNA"</span>, 
<span class="st">"TMEM30A"</span>, <span class="st">"CCNH"</span>, <span class="st">"TRPM8"</span>, <span class="st">"MAOA"</span>, <span class="st">"OPRK1"</span>, <span class="st">"S100A10"</span>, <span class="st">"CUEDC2"</span>, 
<span class="st">"HSD17B4"</span>, <span class="st">"STEAP1"</span>, <span class="st">"PTGFR"</span>, <span class="st">"PCGEM1"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">epic_granges</span><span class="op">)</span>
<span class="va">pidsSE_p</span> <span class="op">=</span> <span class="fu"><a href="../reference/subset_to_promoters.html">subset_to_promoters</a></span><span class="op">(</span><span class="va">pidsSE</span><span class="op">)</span>
<span class="va">inlac</span> <span class="op">=</span> <span class="va">epic_granges</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">epic_granges</span><span class="op">$</span><span class="va">in_prom</span> <span class="op">%in%</span> <span class="va">lncap_activ</span><span class="op">)</span><span class="op">]</span>
<span class="va">ok</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">inlac</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">pidsSE_p</span><span class="op">)</span><span class="op">)</span>
<span class="va">inlacSE</span> <span class="op">=</span> <span class="va">pidsSE_p</span><span class="op">[</span><span class="va">ok</span>,<span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">inlacSE</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/make.names.html">make.names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html">rowRanges</a></span><span class="op">(</span><span class="va">inlacSE</span><span class="op">)</span><span class="op">$</span><span class="va">symbol</span>, unique<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">olac</span> <span class="op">=</span> <span class="fu"><a href="../reference/rept_filtse.html">rept_filtse</a></span><span class="op">(</span><span class="va">inlacSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">olac</span><span class="op">)</span></code></pre></div>
<p><img src="meffilTools_files/figure-html/nxt-1.png" width="700"></p>
<pre><code>## 
## Call:
##  randomForest(formula = y ~ ., data = newdf, importance = TRUE) 
##                Type of random forest: classification
##                      Number of trees: 500
## No. of variables tried at each split: 9
## 
##         OOB estimate of  error rate: 53.33%
## Confusion matrix:
##         CAF Guthrie LNCaP NAF PrEC class.error
## CAF       0       0     0   3    0           1
## Guthrie   0       5     0   0    0           0
## LNCaP     0       0     2   0    0           0
## NAF       3       0     0   0    0           1
## PrEC      0       0     0   2    0           1</code></pre>
<p><img src="meffilTools_files/figure-html/nxt-2.png" width="700"></p>
<p>The marginal distributions of betas in promoter regions of several of these genes clearly distinguish LNCaP.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">boxit</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/boxplot.html">boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html">assay</a></span><span class="op">(</span><span class="va">inlacSE</span><span class="op">[</span><span class="va">x</span>,<span class="op">]</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/substr.html">substr</a></span><span class="op">(</span><span class="va">inlacSE</span><span class="op">$</span><span class="va">Sample_Group</span>,<span class="fl">1</span>,<span class="fl">5</span><span class="op">)</span><span class="op">)</span>, ylab<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/sprintf.html">sprintf</a></span><span class="op">(</span><span class="st">"beta: %s"</span>, <span class="va">x</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="fu">boxit</span><span class="op">(</span><span class="st">"S100A10.3"</span><span class="op">)</span>
<span class="fu">boxit</span><span class="op">(</span><span class="st">"ENO2"</span><span class="op">)</span>
<span class="fu">boxit</span><span class="op">(</span><span class="st">"TMEM30A.7"</span><span class="op">)</span>
<span class="fu">boxit</span><span class="op">(</span><span class="st">"CCNH.6"</span><span class="op">)</span></code></pre></div>
<p><img src="meffilTools_files/figure-html/lkbox-1.png" width="768"></p>
</div>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Min2018" class="csl-entry">
Min, J. L., G. Hemani, G. Davey Smith, C. Relton, and M. Suderman. 2018. <span>“<span class="nocase">Meffil: Efficient normalization and analysis of very large DNA methylation datasets</span>.”</span> <em>Bioinformatics</em> 34 (23): 3983–89. <a href="https://doi.org/10.1093/bioinformatics/bty476">https://doi.org/10.1093/bioinformatics/bty476</a>.
</div>
<div id="ref-Pidsley2016" class="csl-entry">
Pidsley, Ruth, Elena Zotenko, Timothy J. Peters, Mitchell G. Lawrence, Gail P. Risbridger, Peter Molloy, Susan Van Djik, Beverly Muhlhausler, Clare Stirzaker, and Susan J. Clark. 2016. <span>“<span class="nocase">Critical evaluation of the Illumina MethylationEPIC BeadChip microarray for whole-genome DNA methylation profiling</span>.”</span> <em>Genome Biology</em> 17 (1): 1–17. <a href="https://doi.org/10.1186/s13059-016-1066-1">https://doi.org/10.1186/s13059-016-1066-1</a>.
</div>
<div id="ref-Romanuik2010" class="csl-entry">
Romanuik, Tammy L, Gang Wang, Olena Morozova, Allen Delaney, Marco A Marra, and Marianne D Sadar. 2010. <span>“<span class="nocase">LNCaP Atlas: gene expression associated with in vivo progression to castration-recurrent prostate cancer.</span>”</span> <em>BMC Medical Genomics</em> 3 (September): 43. <a href="https://doi.org/10.1186/1755-8794-3-43">https://doi.org/10.1186/1755-8794-3-43</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by VJ Carey.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
